src/components/SortControls.jsx

```diff
@@ -0,0 +1,63 @@
+import {
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  IconButton,
+  Box,
+  Grid,
+} from "@mui/material";
+import ArrowUpwardIcon from "@mui/icons-material/ArrowUpward";
+import ArrowDownwardIcon from "@mui/icons-material/ArrowDownward";
+import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+
+export default function SortControls() {
+  const navigate = useNavigate();
+  const [order, setOrder] = useState("asc");
+
+  function handleSortChange(e) {
+    navigate(`/articles?sort_by=${e.target.value}`);
+  }
+
+  function handleOrderChange() {
+    setOrder((prev) => (prev === "asc" ? "desc" : "asc"));
+    navigate(`/articles?order=${order}`);
```

Comment:
Using useNavigate hook to set up the url is OK, but this way will lose the previous setting. e.g. if the user want to both sort it by votes and ascending order, it won't work. By default it'll jump back to sort by date. Also the url won't show as "/articles?sort_by=votes&order=asc".